### PR TITLE
add 2moons distribution and affine transformation

### PR DESCRIPTION
**Pull request by Justinezgh**
image
image

[notebook](https://github.com/Justinezgh/SBI-Diff-Simulator/blob/61b3f51a42153072ac81bc2c8befe99f19159766/notebooks/smoothNF_VS_affineNF.ipynb)

### CODE REVIEW

File: sbids/tasks/two_moons.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import jax.numpy as np
+from sklearn.datasets import make_swiss_roll
```
Comment (EiffL) [inline sbids/tasks/two_moons.py:2]:
```suggestion
```

**Comment by EiffL** [comment]
Ok, that's great @Justinezgh ! Now, what we want is for this notebook to produce a beautiful plot that we can put in the paper and future presentations.

**Comment by EiffL** [comment]
Kind of like the plots we had there: https://eiffl.github.io/Paris2022/#/30/0/0

File: notebooks/smoothNF_VS_affineNF.ipynb [hunk 2]
```diff
@@ -0,0 +1,951 @@
...
+            "Requirement already satisfied: decorator in /usr/local/lib/python3.7/dist-packages (from tensorflow_probability>=0.14.1->SBI-Diff-Simulator==0.0.1) (4.4.2)\n",
+            "Requirement already satisfied: cloudpickle>=1.3 in /usr/local/lib/python3.7/dist-packages (from tensorflow_probability>=0.14.1->SBI-Diff-Simulator==0.0.1) (1.3.0)\n",
+            "Requirement already satisfied: gast>=0.3.2 in /usr/local/lib/python3.7/dist-packages (from tensorflow_probability>=0.14.1->SBI-Diff-Simulator==0.0.1) (0.5.3)\n"
+          ]
+        }
+      ],
+      "source": [
+        "!pip install --quiet --upgrade dm-haiku optax \n",
+        "!pip install git+https://github.com/Justinezgh/SBI-Diff-Simulator.git\n",
+        "!pip install --quiet git+https://github.com/astrodeepnet/sbi_experiments.git"
```
Comment (EiffL) [inline notebooks/smoothNF_VS_affineNF.ipynb:74]: Why do we need sbi_experiments?

Comment (Justinezgh) [reply notebooks/smoothNF_VS_affineNF.ipynb:74]: we don't, forgot to change it

**Review by EiffL — CHANGES_REQUESTED** [review]
My main comment is to use this notebook to make publishable plots. You can even create a folder "figures" in which you store the figures that this notebook generates, and these plots are intended to go directly into a paper.

Can you update the notebook accordingly? 

Once this is done, this is done and we can focus on something else

**Comment by Justinezgh** [comment]
image
image

**Comment by EiffL** [comment]
Tres jolie :-) But what I was saying last night, is that instead of sampling from the points you get the gradients for from the true distribution, maybe we want to sample the points from the **learned** distribution. Have you tried that? Is it not looking good?
We also need a plot of the true score for the true distribution :-)

**Comment by EiffL** [comment]
yep :-) that looks good I think

**Comment by Justinezgh** [comment]
ok cool :) 


I ran the 2moons experiments on azure today, for each nb of simulations 10 NFs (with different initialization seed) are trained 
image
image

**Comment by aboucaud** [comment]
Looks really good. Especially the bump that we saw previously at ~200 sims seems to have disappeared.

**Comment by EiffL** [comment]
:-D makes me very happy. One comment and a question:
- Maybe on the x axis here we could say, "number of training samples", here we are not really doing SBI, it's just density estimation, so the word simulation could be misleading. On the first plot, on the y axis it says `nlp` when I think you want to say `NLL` in this particular case.

- Just a question (not saying we should do it necessarily), have you tried training just with the score and see what these plots would look like?

**Review by EiffL — APPROVED** [review]
LGTM, I'd just recommend also including the scripts/notebook used to generate these nice plots as a fucntion of number of simulations

**Comment by Justinezgh** [comment]
No I haven't try with just the score but I can

**Comment by EiffL** [comment]
Ok, not good I guess. Probably this normalization question we talked about. It would mess with both metrics

**Comment by Justinezgh** [comment]
@EiffL can I merge ?

**Comment by EiffL** [comment]
All good for me
